### PR TITLE
PYTHON-432: Add IF EXISTS support to UPDATE and DELETE statements of CQLEngine

### PR DESCRIPTION
**Pull request by aboudreault**

### CODE REVIEW

**Comment by aboudreault** [comment]
Added some documentation.

File: docs/api/cassandra/cqlengine/models.rst [hunk 1]
```diff
@@ -82,7 +82,7 @@ Model
 
     .. automethod:: create
 
-    .. method:: if_not_exists()
+    .. method:: if_not_exists(boolean)
```
Comment (aholmberg) [inline docs/api/cassandra/cqlengine/models.rst]: I don't think this is right. It should not take an argument.

Comment (aboudreault) [inline docs/api/cassandra/cqlengine/models.rst]: This issue is related to the fact that these settings use Descriptor.

File: docs/api/cassandra/cqlengine/models.rst [hunk 2]
```diff
@@ -100,6 +100,24 @@ Model
...
+        Check the existence of an object before an update or delete. The existence of an
+        object is determined by its primary key(s). And please note using this flag
+        would incur performance cost.
+
+        If the update or delete isn't applied, a :class:`~cassandra.cqlengine.query.LWTException` is raised.
+
+        .. code-block:: python
+
+            try:
+                TestIfExistsModel.if_exists(id=id).update(count=9, text='111111111111')
```
Comment (aholmberg) [inline docs/api/cassandra/cqlengine/models.rst]: I don't think if_exists accepts kwargs (?)

Comment (aboudreault) [inline docs/api/cassandra/cqlengine/models.rst]: this is wrong though, will fix

File: docs/api/cassandra/cqlengine/models.rst [hunk 3]
```diff
@@ -100,6 +100,24 @@ Model
 
         This method is supported on Cassandra 2.0 or later.
 
+    .. method:: if_exists(boolean)
```
Comment (aholmberg) [inline docs/api/cassandra/cqlengine/models.rst]: Does not take boolean

File: docs/api/cassandra/cqlengine/models.rst (hunk 1 above)
Comment (aholmberg) [inline docs/api/cassandra/cqlengine/models.rst]: I see. Sorry for the confusion. I think we should try to get away from that inconsistency. Maybe we can default that parameter ``True`` in the setter and take it away later.

**Comment by aboudreault** [comment]
Addressed comments.

**Pull request merged**